content/en/platform/corda/5.0-dev-preview-2/getting-started/first-flow/code-java.md

```diff
@@ -0,0 +1,458 @@
+```
+## Setting up the FlowSession
+We can now start sending messages to the responder:
+1. Set up a `FlowSession` between the initiator and responder node:
+   ```java
+        FlowSession session = flowMessaging.initiateFlow(otherMember);
+   ```
+2. Simply call `send()` on the session and pass a payload:
+   ```java
+        session.send(message);
+   ```
+   The code continues to execute until it reaches the `session.receive()` method. At that point, the flow checkpoints and persists its state to the database. It resumes when it receives a message back from the responder. This frees up the Corda cluster flow workers to perform other tasks.
+   {{< note >}}
+   There is no guarantee that the same flow worker resumes the completion of the flow and so singleton objects should be avoided in Corda 5 flows.
+   {{< /note >}}
+   ```java
+           Message response = session.receive(Message.class);
+    ```
+    When the `send()` is sent to the responder, Corda executes the `MyFirstResponderFlow` responder flow `call()` method down to the `session.receive()`, which returns the payload that was sent in the initiator's `send()` method.
+    ```java
+    @Suspendable
+    @Override
+    public void call(FlowSession session) {
+      log.info("MFF: MyFirstResponderFlow.call() called");
+      Message receivedMessage = session.receive(Message.class);
+      ...
+    }
+    ```
+
+   The responder flow then obtains its own identity from its injected `memberLookup` service, creates a response, and sends the response back using `session.send()`:
+   ```java
+      MemberX500Name ourIdentity = memberLookup.myInfo().getName();
+      Message response = new Message(ourIdentity,
+             "Hello " + session.getCounterparty().getCommonName() + ", best wishes from " + ourIdentity.getCommonName());
+     log.info("MFF: response.message: " + response.message);
+     session.send(response);
+   ```
+   There is no return value for a ResponderFlow.
+
+   Back on the Initiating virtual node, Corda detects the send coming from the responder and rehydrates the initiating flow from the database and resumes it from where it was checkpointed.
+
+   The response is received and `response.message` is returned by the flow.
+   ```java
+      Message response = session.receive(Message.class);
+      return response.message;
+   ```
+   The response from the initiating flow is always a string, which can be returned when the flow status is queried by HTTP-RPC.
+
+## Other Considerations for FlowSessions TO BE UPDATED
+It is important that the sends and receives in the initiator and responder flows match. If the initiator sends a Foo and the responder expects a Bar, the flow hangs and likely results in a timeout error.
+
+As with Corda 4, there is also a `sendAndReceive` method on `FlowSession` that sends a payload, check-points the flow, and then waits for a response to be received:
+```kotlin
+val response = sendAndReceive(<ReceiveType>::class.java, payload)
```

Comment:
In Java this should be:
\<ReceiveType\> response = sendAndReceive(\<ReceiveType\>.class, payload);